feat: get apps from backend

laurafitzgerald opened the pull request:
## Motivation
https://issues.jboss.org/browse/AEROGEAR-8376

## What
- Added DataService and fetch.js
- Created 3 new reducers to deal with the fetch action, success, failure
- Added boolean state object didfetchfail
- Added logic to render the table on didfetchfail false else render message `unable to fetch apps` - we may want to clean that up to do a nice blank slate experience

## Why
Satisfy AC 

## How
See What

## Verification Steps
_Verify Failed Request on no Db running_
1. Stop the DB if you have it running
2. Run `make serve`
3. Verify that the message `Unable to fetch any apps` is shown

_Verify Failed Request on no returned apps_
1. Run the db using `docker-compose up db` as described [here](https://github.com/aerogear/mobile-security-service#16-running-entire-application-with-docker-compose)
2. Check that there are no records in the db
3. Verify that the message `Unable to fetch any apps` is shown

_Verify Success Request_

1. Run make test-integration to seed the database
2. Refresh the UI - Verify the UI appears as shown below

![screenshot 2019-03-01 at 11 27 07](https://user-images.githubusercontent.com/6498727/53640817-12e58900-3c25-11e9-8822-8c049af3d406.png)

_Run Tests_

1. Run `npm run test`
2. All tests should pass

## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Progress

- [x] Functionality
- [x] clean up on the request as the link is currently hardcoded
- [x] write tests for the new actions/reducers/ui

laurafitzgerald commented on ui/src/DataService.js:
```diff
@@ -0,0 +1,46 @@
+});
+
+const request = async (url, method, body) => {
+  const response = await fetch(`http://localhost:3000${baseUrl}/${url}`, requestConfig(method, body));
```
This needs to be updated to not use the hardcoded url but pushing now in order to show the redux state management is working first.

laurafitzgerald commented on ui/src/reducers/index.js:
```diff
@@ -35,6 +29,29 @@ export default (state = initialState, action) => {
+      var fetchedApps = [];
+      for (var i = 0; i < action.result.length; i++) {
+        var temp = [];
+        temp[0] = action.result[i].appId;
```
This might not be the best way to do this. It's a bit rough and ready still.

damienomurchu replied:
You should be able to replace the `for` loop with `map()` on the array if you wanted, but the explicit code looks fine to me

laurafitzgerald replied:
yeah I used my java knowledge here, I knew it wasn't ideal. Going to look into `map()` with js.

damienomurchu replied:
tbh, may not really be too many gains there, but might be slightly more readable.

laurafitzgerald commented:
@damienomurchu @craicoverflow pushing this early version of the implementation of getting the apps. I've left comments above on pieces I know need to be updated but pushing this working implementation to get early feedback on the approach. Going to progress to adding some testing while I wait for feedback from you both around the current bits.

damienomurchu commented:
Functionality looks good. Data service implementation as per MDC and looks fine. Will the clarification we're awaiting from Chris on trello block this task?

laurafitzgerald commented:
Thanks @damienomurchu I think we can create a new task for that rather than blocking this one. For now I've improved the loop. I will start writing tests for the new reducers and actions now. Also there's some changes coming from @davidffrench around running the server and UI which may impact this one also.

davidffrench commented:
> Thanks @damienomurchu I think we can create a new task for that rather than blocking this one. For now I've improved the loop. I will start writing tests for the new reducers and actions now. Also there's some changes coming from @davidffrench around running the server and UI which may impact this one also.

Those changes are already merged into master. It is just a `make serve` command. This allows the UI to come from the Go server so they will both be running on the same port. This will not work for development but can be apart of the verification steps now to verify a ticket.

laurafitzgerald commented:
@davidffrench @craicoverflow I've removed the hardcoded localhost url now. This now works with `make serve`. Would you mind verifying this now? 
I've updated the verification steps in the description

laurafitzgerald commented:
All code and test completed. Request for review please @davidffrench @craicoverflow @damienomurchu

austincunningham commented:
## Verification
_Verify Failed Request on no Db running_

:-1:  not a valid verification step as go server falls over if no db, as discussed

_Verify Failed Request on no returned apps_
:+1: 
![image](https://user-images.githubusercontent.com/16667688/53815705-0a10f200-3f5a-11e9-93cb-b3ad4e64a4fa.png)


_Verify Success Request_
:+1: 

_Run Tests_
```
 PASS  src/actions/__tests__/actions-ui.test.js
 PASS  src/reducers/__tests__/index.test.js
 PASS  src/containers/__tests__/AppsTableContainer.test.js
 PASS  src/components/__tests__/App.test.js
 PASS  src/components/__tests__/AppDetailedView.test.js
 PASS  src/components/__tests__/AppsTable.test.js

Test Suites: 6 passed, 6 total
Tests:       10 passed, 10 total
Snapshots:   0 total
Time:        4.39s
Ran all test suites.

Watch Usage: Press w to show more.
```

## Verified
